cmd/helmExecute.go

```diff
@@ -120,3 +130,77 @@ func runHelmExecuteDefault(config helmExecuteOptions, helmExecutor kubernetes.He
 
 	return nil
 }
+
+// parseAndRenderCPETemplate allows to parse and render a template which contains references to the CPE
+func parseAndRenderCPETemplate(config helmExecuteOptions, rootPath string, utils kubernetes.DeployUtils) error {
+	cpe := piperenv.CPEMap{}
+	err := cpe.LoadFromDisk(path.Join(rootPath, "commonPipelineEnvironment"))
+	if err != nil {
+		return fmt.Errorf("failed to load values from commonPipelineEnvironment: %v", err)
+	}
+
+	valueFiles := []string{}
+	defaultValueFile := fmt.Sprintf("%s/%s", config.ChartPath, "values.yaml")
```

Comment:
There's one more thing that just came to my mind. Making the default `values.yaml` file a template is probably a bad idea, because such a chart would not be usable outside of the Piper context.
On the other hand side, any additional values files backed into the chart are not accessible without downloading the Helm chart and extracting the value file.

Not sure how useable this implementation will turn out to be. Maybe it would be better to treat the value file with default images as a separate artefact? Not sure how it would be linked to the chart artefact though either.

Reply:
Correct me if I'm wrong, we have a repository with application, Dockerfile, and helm chart directory with template values in values files. We build application, docker image, and then the helmExecute step retrieves image-related information and inserts/renders it to values files. Then the helmExecute, for example, packages helm chart and publishes artifact with concrete values in values file(s) to storage and we can use artifact whenever we want. Isn't?

If we don’t want to use template values in helm chart, we can just provide concrete values.

Reply:
The point I was trying to make was, that it might be difficult to use this implementation as a feature.

Of course, one does not have to provide a template, the Go templating engine will accept any file without template syntax just fine and do a no-op.
It's a goal conflict imho. You would want to have a literal values file locally, so that you can call e.g. `helm template` to check the templates are fine, you would want to run `helm lint` and potentially even test deploy with `helm install` or `helm upgrade -I` - and it's ok to not have concrete image information. In your CI though, you'd want to have the concrete images provided with maximum convenience and you would only get that, if you'd put the image information into the default `values.yaml`.

The current implementation doesn't provide both from my POV.

Reply:
for the current implementation its fine if we provide a piper specific template , since currently we would need some common mechanism to update the values. updated Charts are uploaded to a registry and can be downloaded if need be to use them outside piper